### PR TITLE
Assorted improvements following up #6658

### PR DESCRIPTION
**Pull request by nicoddemus**

### CODE REVIEW

**Comment by nicoddemus** [comment]
> In general I've started to play around with making the terminal writer handle a pygments Token stream, which allows for e.g. LineMatcherFailed exceptions to be colored / tokenized.

Definitely! We should use this for all custom failure exceptions (text diff, list diffs, etc) like has been done in [ward](https://wardpy.com/).

Interesting idea of supporting pygments token streams directly in our subclass, had not thought about that.

**Comment by nicoddemus** [comment]
> Interesting idea of supporting pygments token streams directly in our subclass, had not thought about that.

We can probably see how popular the color feature is (I suspect huge), so we can then see if there are problems on depending on pygments directly (I think none).

File: src/_pytest/_code/code.py [hunk 1]
```diff
@@ -1048,28 +1048,35 @@ def _write_entry_lines(self, tw: TerminalWriter) -> None:
         character, as doing so might break line continuations.
         """
 
-        indent_size = 4
```
Comment (RonnyPfannschmidt) [inline src/_pytest/_code/code.py]: that diff was fun, without expanding it looks ike a api change

**Comment by blueyed** [comment]
@nicoddemus I will not approve/comment here (in case you are waiting) - the whole highlighting was very much premature IMHO, and fixes like this only cause conflicts for me.
(it is still good in general given the current situation here though)

**Pull request merged**